pkg/apis/camel/v1/integration_types_support.go

```diff
@@ -25,8 +25,18 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// IntegrationLabel is used to tag k8s object created by a given Integration.
 const IntegrationLabel = "camel.apache.org/integration"
 
+// IntegrationSyntheticLabel is used to tag k8s synthetic Integrations.
+const IntegrationSyntheticLabel = "camel.apache.org/is-synthetic"
+
+// IntegrationImportedKindLabel specifies from what kind of resource an Integration was imported.
+const IntegrationImportedKindLabel = "camel.apache.org/imported-from-kind"
```

Comment:
maybe we can use ownership here  so the resource is also the owner of the synthetic integration and can get deleted by kubernetes gc ? it could also simplify the reconciler as you probably won't need to delete the resource

Reply:
The reconciliation deletion should still exists, when the user remove the importing label because he wants the Deployment to no longer generates a synthetic Integration.

Reply:
> I thought that once the user imports an Integration, he wants to control the lifecycle as well

I think it would become very complicated to do it right. In my opinion, the synthetic integration is essentially a read only projection of the status of an camel application so I won't go much further.

> The reconciliation deletion should still exists, when the user remove the importing label because he wants the Deployment to no longer generates a synthetic Integration.

true, but I guess if this should be something that falls in a  GC algorithm as I don't know if the way it is implemented as today, then it would work in case of a restart of the operator as you may not receive a deletion event (but have had a look at the code in depth so I may be wrong).

Reply:
Done.